src/API.jl

```diff
@@ -69,13 +69,27 @@ end
 
 macro odbc(func,args,vals...)
     esc(quote
+        ret = SQL_SUCCESS
         if odbc_dm[] == iODBC
-            ccall( ($func, iODBC_dm), SQLRETURN, $(swapsqlwchar(args)), $(vals...))
+            while true
+                ret = ccall( ($func, iODBC_dm), SQLRETURN, $(swapsqlwchar(args)), $(vals...))
+                ret == SQL_STILL_EXECUTING || break
+                sleep(0.001)
```

Comment:
Oh man, not sure that's actually better. Might be better just to call `@spawn` and hope there are real threads available. Is there some way to get the driver to call back when there's data available?

Reply:
There is, but it's even more rarely implemented than this form of "asyncness". It's also a lot more complicated, so I'm not super excited for all the cfunction/ccall munging it would take to get a feature that may be a database or two bothered to implement.

Do you think it'd be better to bump up the sleep time here? Would that be slightly better?

Reply:
This is probably ok. It's enough time to yield a bit and let some other work get done often enough. `sleep` only sets a lower bound on how long you sleep, not an upper bound.